pkg/operator/target_config_reconciler.go

```diff
@@ -1070,8 +1069,10 @@ func (c *TargetConfigReconciler) manageMetricsCertificateCR(ctx context.Context,
 			},
 			"spec": map[string]interface{}{
 				"dnsNames": []interface{}{
-					metricsServiceName,
+					fmt.Sprintf("kueue-controller-manager-metrics-service.%s.svc", c.operatorNamespace),
```

Comment:
nit:
You can use `fmt.Sprintf` once like metricsServiceName := `fmt.Sprintf("kueue-controller-manager-metrics-service.%s.svc", c.operatorNamespace)`
and in `dnsNames`
```go
				"dnsNames": []interface{}{
					metricsServiceName,
					metricsServiceName + ".cluster.local",
				},
```